File: ReactApp/app/types/interviews/interview.ts
export interface Interview {
    InterviewID: number;
    InterviewEntryDate: string | null; // Updated to allow null (ISO date or null)
    RecruiterID: number | null;
    InterviewDate: string | null; // ISO date or null
    InterviewStartTime: string | null;
    InterviewEndTime: string | null;
    ClientID: number | null;
    InterviewType: string | null;
    InterviewMethod: string | null;
    Technology: string | null;
    InterviewFeedback: string | null;
    InterviewStatus: string | null;
    InterviewSupport: string | null;
    Comments: string | null;
    CreatedDate: string | null; // ISO datetime (e.g., "2025-02-28T00:00:00Z")
    ModifiedDate: string | null; // ISO datetime
    EndClientName: string | null;
  }

File: Backend/Models/Interview.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Backend.Models
{
    public class Interview
    {
        [Key]
        public int InterviewID { get; set; }

        [Required]
        public DateTime InterviewEntryDate { get; set; }

        public int? RecruiterID { get; set; }
        [ForeignKey(nameof(RecruiterID))]
        public Employee? Recruiter { get; set; }

        public DateTime? InterviewDate { get; set; }

        [MaxLength(50)]
        public string? InterviewStartTime { get; set; }

        [MaxLength(50)]
        public string? InterviewEndTime { get; set; }

        public int? ClientID { get; set; }
        [ForeignKey(nameof(ClientID))]
        public Client? Client { get; set; }

        [MaxLength(50)]
        public string? InterviewType { get; set; }

        [MaxLength(50)]
        public string? InterviewStatus { get; set; }
        [MaxLength(50)]
        public string? InterviewMethod { get; set; }
        [MaxLength(50)]
        public string? Technology { get; set; }
        [MaxLength(50)]
        public string? InterviewFeedback { get; set; }

        public string? InterviewSupport { get; set; }

        public bool? IsActive { get; set; }

        public string? Comments { get; set; }
        public DateTime CreatedDate { get; set; } = DateTime.UtcNow;
        public DateTime ModifiedDate { get; set; } = DateTime.UtcNow;

        public string? EndClientName { get; set; }
    }
}